Description: Dynamically link ppdc, to work around segfault on mipsen.
Author: Sune Vuorela <debian@pusling.com>
Bug-Debian: http://bugs.debian.org/548246
Last-Update: 2015-02-10

--- a/ppdc/Makefile
+++ b/ppdc/Makefile
@@ -236,14 +236,14 @@
 # genstrings - generate GNU gettext strings.
 #
 
-genstrings:		genstrings.o libcupsppdc.a ../cups/$(LIBCUPSSTATIC) \
+genstrings:		genstrings.o libcupsppdc.so ../cups/$(LIBCUPSSTATIC) \
 			sample.drv ../data/media.defs
 	echo Linking $@...
 	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o genstrings genstrings.o \
-		libcupsppdc.a ../cups/$(LIBCUPSSTATIC) $(LIBGSSAPI) $(SSLLIBS) \
+		-lcupsppdc ../cups/$(LIBCUPSSTATIC) $(LIBGSSAPI) $(SSLLIBS) \
 		$(DNSSDLIBS) $(COMMONLIBS) $(LIBZ)
 	echo Generating localization strings...
-	./genstrings >sample.c
+	LD_LIBRARY_PATH=.:../cups/ ./genstrings >sample.c
 
 
 #
@@ -257,12 +257,12 @@
 
 ppdc-static:		ppdc.o libcupsppdc.a ../cups/$(LIBCUPSSTATIC) foo.drv foo-fr.po
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o ppdc-static ppdc.o libcupsppdc.a \
+	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o ppdc-static ppdc.o -lcupsppdc \
 		../cups/$(LIBCUPSSTATIC) $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
 		$(COMMONLIBS) $(LIBZ)
 	echo Testing PPD compiler...
-	./ppdc-static -l en,fr -I ../data foo.drv
-	./ppdc-static -l en,fr -z -I ../data foo.drv
+	LD_LIBRARY_PATH=.:../cups/ ./ppdc-static -l en,fr -I ../data foo.drv
+	LD_LIBRARY_PATH=.:../cups/ ./ppdc-static -l en,fr -z -I ../data foo.drv
 
 
 #
@@ -285,14 +285,14 @@
 
 ppdi-static:		ppdc-static ppdi.o libcupsppdc.a  ../cups/$(LIBCUPSSTATIC)
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o ppdi-static ppdi.o libcupsppdc.a \
+	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o ppdi-static ppdi.o -lcupsppdc \
 		../cups/$(LIBCUPSSTATIC) $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
 		$(COMMONLIBS) $(LIBZ)
 	echo Testing PPD importer...
 	$(RM) -r ppd ppd2 sample-import.drv
-	./ppdc-static -l en -I ../data sample.drv
-	./ppdi-static -I ../data -o sample-import.drv ppd/*
-	./ppdc-static -l en -I ../data -d ppd2 sample-import.drv
+	LD_LIBRARY_PATH=.:../cups/ ./ppdc-static -l en -I ../data sample.drv
+	LD_LIBRARY_PATH=.:../cups/ ./ppdi-static -I ../data -o sample-import.drv ppd/*
+	LD_LIBRARY_PATH=.:../cups/ ./ppdc-static -l en -I ../data -d ppd2 sample-import.drv
 	if diff -r ppd ppd2 >/dev/null; then \
 		echo PPD import OK; \
 	else \
@@ -325,7 +325,7 @@
 
 testcatalog:		testcatalog.o libcupsppdc.a ../cups/$(LIBCUPSSTATIC)
 	echo Linking $@...
-	$(CXX) $(LDFLAGS) -o $@ testcatalog.o libcupsppdc.a \
+	$(CXX) $(LDFLAGS) -o $@ testcatalog.o -lcupsppdc \
 		../cups/$(LIBCUPSSTATIC) $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
 		$(COMMONLIBS) $(LIBZ)
 
